src/distance/ConfigurationRanker.cpp

```diff
@@ -25,6 +28,49 @@ ConfigurationRanker::ConfigurationRanker(
           mMetaSkeletonStateSpace));
 }
 
+//==============================================================================
+ConfigurationRanker::ConfigurationRanker(
+    ConstMetaSkeletonStateSpacePtr metaSkeletonStateSpace,
+    ConstMetaSkeletonPtr metaSkeleton,
+    std::vector<double> weights)
+  : mMetaSkeletonStateSpace(std::move(metaSkeletonStateSpace))
+  , mMetaSkeleton(std::move(metaSkeleton))
+{
+  if (!mMetaSkeletonStateSpace)
+    throw std::invalid_argument("MetaSkeletonStateSpace is nullptr.");
+
+  if (!mMetaSkeleton)
+    throw std::invalid_argument("MetaSkeleton is nullptr.");
+
+  if (mMetaSkeletonStateSpace->getDimension() != weights.size())
+    throw std::invalid_argument(
+        "Weights should have the same dimension as the "
+        "MetaSkeletonStateSpace.");
+
+  for (std::size_t i = 0; i < weights.size(); ++i)
+  {
+    if (weights[i] < 0)
+      throw std::invalid_argument("Weights should all be non-negative.");
+  }
+
```

Comment:
Should we normalize it to 1  somewhere here?

Reply:
I dont think it matters